code/core/src/manager/components/sidebar/TagsFilter.stories.tsx

```diff
@@ -0,0 +1,61 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import { findByRole, fn } from '@storybook/test';
+
+import { TagsFilter } from './TagsFilter';
+
+const meta = {
+  component: TagsFilter,
+  tags: ['haha'],
+  args: {
+    api: {
+      experimental_setFilter: fn(),
+      getDocsUrl: () => 'https://storybook.js.org/docs/',
+    } as any,
+    updateQueryParams: fn(),
+  },
+} satisfies Meta<typeof TagsFilter>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Closed: Story = {
+  args: {
+    indexJson: {
+      v: 6,
+      entries: {
+        'c1-s1': { tags: ['A', 'B', 'C', 'dev'] } as any,
+      },
+    },
+  },
+};
+
+export const ClosedWithSelection: Story = {
+  args: {
+    ...Closed.args,
+    initialSelectedTags: ['A', 'B'],
+  },
+};
+
+export const Open: Story = {
+  ...Closed,
+  play: async ({ canvasElement }) => {
+    const button = await findByRole(canvasElement, 'button');
+    await button.click();
+  },
+};
+
+export const OpenWithSelection: Story = {
+  ...ClosedWithSelection,
+  play: Open.play,
+};
+
+export const OpenEmpty: Story = {
```

Comment:
I would if there shoould also be an empty state for when only internal tags are present, and users haven't yet defined custom tags? You mentioned wanting to show the `play-fn` tag but it would be a shame if it got in the way of feature onboarding.

Reply:
Good call